### PR TITLE
Added some very ugly deprecation warnings to some key files

### PR DESCRIPTION
**Pull request by perlun**
These files represent (to the best of my knowledge) some of our most common parts of the 'facets' gem. The idea: because it is known to cause conflicts with the `activesupport` gem (seen as late as today with @mriska and @tforsbacka), let us deprecate all of this as far as our platform is concerned, so we can get see where it is being used and eventually get rid of the whole thing, in a controlled manner.

Trust me: I know that global variables are bad. The idea here is to set a flag to avoid more warnings than is needed. I.e. if the user/an app is loading 'facets', we only need to produce a single warning for that - no need to warn multiple times. Maybe that's something we could live with, but... I'd rather not spend more time on this now since it's legacy cruft (from the perspective of our platform) so I don't think 100% pureness is needed in this case.

@jensnockert - you seem to be online. Care to dig out your review skills?

### CODE REVIEW

**Comment by perlun** [comment]
Ping Jens - did you see this stuff also?

File: lib/core/facets.rb [hunk 1]
```diff
@@ -1,3 +1,10 @@
+unless $FACETS_LOADED
```
Comment (jesjos) [inline lib/core/facets.rb]: Just reading this the first time, the naming confused me.

Perhaps `$IGNORE_FACETS_WARNING` or something else would be more descriptive?

I think the global variable could be replaced with something like this:

``` ruby
module Ecraft
  module Facets
    class << self
      attr_reader :already_warned
    end
  end
end

unless Ecraft::Facets.already_warned
  # deprecation code
end
```

Comment (perlun) [inline lib/core/facets.rb]: Yeah, the naming is very lame here. I think your suggestion is good - will apply it. (I just did it in a slightly more "sloppy" fashion this time, given the anticipated short-lived nature of this fork.)

Comment (perlun) [inline lib/core/facets.rb]: However, that means I would have to put this in a shared file being loaded by the different files (since we don't know that the user will `require 'facets`'; they could as well be requiring `facets/foo` - so I cannot just put that in `facets.rb`)

I think I will go the KISS approach and remove the global altogether. It's not a big deal if the user is warned a couple of times in this case methinks.

**Comment by perlun** [comment]
Can I merge this now?

**Comment by olleolleolle** [comment]
`:require_facets` is now in the list of Deprecations in the wiki.

https://github.com/ecraft/ecraft.uxfactory.server/wiki/Deprecations

Can you summarize how to upgrade away from the deprecation? In this wiki article in the Deprecations page:  https://github.com/ecraft/ecraft.uxfactory.server/wiki/Deprecation-require_facets

**Comment by perlun** [comment]
Added some docs now. Also removed a deprecation in the server: https://github.com/ecraft/ecraft.uxfactory.server/pull/298

OK to merge now? Or do you want more docs?

**Comment by olleolleolle** [comment]
:+1: r=@olleolleolle

Great work!

**Comment by perlun** [comment]
Thanks Olle! Merging now.
